src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -81,12 +87,13 @@ public class StoreService : IStoreService
             {
                 _dicomInstanceEntries = instanceEntries;
                 _requiredStudyInstanceUid = requiredStudyInstanceUid;
+                List<CustomTagEntry> customTagEntries = (await _customTagStore.GetCustomTagsAsync(path: null, cancellationToken: cancellationToken)).ToList();
```

Comment:
Instead of making this a list, can we simply return some sort of `List<T>`-like type from `GetCustomTagsAsync`?